controller/tenant.go

```diff
@@ -68,42 +65,32 @@ func (c *TenantController) Setup(ctx *app.SetupTenantContext) error {
 	}
 
 	// fetch the cluster the user belongs to
-	user, err := c.userService.GetUser(ctx, ttoken.Subject())
+	user, err := c.authClientService.NewUser(ctx)
 	if err != nil {
```

Comment:
this is `GetUser()` rather `NewUser()`
and I guess, taking ctx and token as input argument would be better.  Input to method clarify what all it depends on.  In go ctx is such a object where you can add param and pass only ctx around but that's not right.

Reply:
> this is GetUser() rather NewUser()

yeah, I wasn't sure about this name. it can be `GetUser()` because it gets info about user from the auth sevice, but can be `NewUser()` because it constructs a new instance of `auth.User` where the params are taken from more sources.

> and I guess, taking ctx and token as input argument would be better. Input to method clarify what all it depends on. In go ctx is such a object where you can add param and pass only ctx around but that's not right.

well, it depends on the token that is bundled inside of the context of an incoming request. The original code was just doing the same thing (retrieving token from context) on several places when it can be moved directly to the method and can be hidden as this is just an implementation detail. I know that the code of retrieving the token information left in the original places, but will be removed in the future.

I wrote the function with the idea "Create me an instance of the `auth.User` for the given context, where the user represents the client who sent the request"

Reply:
It's definitely GetUser() not NewUser() :) You are calling an auth service to get the existing user. Not to create a new User in the Auth service. The fact that it's creating a new instance from auth.User struct is just an implementation detail of the service. 

Regarding context and token... I agree with @MatousJobanek. We just pass the incoming contest to the service. Extracting the token (which was added there by goa JWT middleware) and creating a new request for Auth is an implementation detail of the service.

Reply:
renamed in 2547d6c